disnake/ext/commands/cog.py

```diff
@@ -416,9 +419,12 @@ def listener(cls, name: str = MISSING) -> Callable[[FuncT], FuncT]:
             The function is not a coroutine function or a string was not passed as
             the name.
         """
-        if name is not MISSING and not isinstance(name, str):
+        if name is not MISSING and not (
+            issubclass(name.__class__, _EnumValueBase) or isinstance(name, str)
+        ):
```

Comment:
```suggestion
        if name is not MISSING and not isinstance(name, (str, Event)):
```
(this change also requires imports to be moved, same thing in other places where this was added)

---
`isinstance(name, Event)` is fine for these checks, thanks to `__instancecheck__`:
https://github.com/DisnakeDev/disnake/blob/13550a10e936671ce3d7ec532e99955c790f368d/disnake/enums.py#L174-L180